charts/everest/charts/common/templates/_psp_cleanup.yaml.tpl

```diff
@@ -0,0 +1,77 @@
+#
+# @param .namespace     The namespace where Everest server is installed
+#
+{{- define "everest.pspCleanup" }}
+{{- $hookName := printf "everest-helm-psp-pre-delete-hook" }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ $hookName }}
+  namespace: {{ .namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ $hookName }}
+  namespace: {{ .namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+rules:
+  - apiGroups:
+      - everest.percona.com
+    resources:
+      - podschedulingpolicies
+    verbs:
+      - get
+      - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ $hookName }}
+  namespace: {{ .namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ $hookName }}
+subjects:
+  - kind: ServiceAccount
+    name: {{ $hookName }}
+    namespace: {{ .namespace }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ $hookName }}-{{ randNumeric 6 }}
+  namespace: {{ .namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    spec:
+      containers:
+        - image: bitnami/kubectl:latest
+          name: {{ $hookName }}
+          command:
+            - /bin/sh
+            - -c
+            - |
+              for pspName in everest-default-pxc everest-default-psmdb everest-default-postgresql
```

Comment:
If at some point we're going to provide more defaults or the default name gets changed we would need to change this loop too. Would it be too bad if we patch all the psps? Besides, users might have applied the `everest.percona.com/readonly-protection` finalizer to other psp manually to protect them.

Reply:
done